pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java

```diff
@@ -227,26 +227,28 @@ public boolean equals(Object obj) {
   }
 
   private boolean compareMap(Map<String, Object> thisMap, Map<String, Object> thatMap) {
-    if (thisMap.size() == thatMap.size()) {
-      for (String key : thisMap.keySet()) {
-        Object fieldValue = thisMap.get(key);
-        Object thatFieldValue = thatMap.get(key);
-        if (fieldValue == null) {
-          if (thatFieldValue != null) {
-            return false;
-          }
-        } else if (!fieldValue.equals(thatFieldValue)) {
-          if (fieldValue instanceof Map && thatFieldValue instanceof Map) {
-            return compareMap((Map<String, Object>) fieldValue, (Map<String, Object>) thatFieldValue);
-          }
-          if ((fieldValue instanceof byte[]) && (thatFieldValue instanceof byte[])) {
-            return Arrays.equals((byte[]) fieldValue, (byte[]) thatFieldValue);
-          }
-          if (fieldValue.getClass().isArray() && thatFieldValue.getClass().isArray()) {
-            return compareArray((Object[]) fieldValue, (Object[]) thatFieldValue);
-          }
+    if (thisMap.size() != thatMap.size()) {
```

Comment:
nit: shall we do null check first?

Reply:
is it possible for these to be null? If os, what would the desired behavior be here? Only return true if they're both null?

Reply:
Right. Maybe reuse this helper method once it’s merged ?
https://github.com/apache/incubator-pinot/pull/6965/files#diff-a5239a99ae5c9433e0bc5fdd5d7b67e45e0f90607d9d21821d24f334bda27ca3R92

Reply:
Oh, that looks perfect and I can totally remove `compareMap`. What do you think the ETA to merge 6595 is?

Reply:
it's merged already, please ensure that method can handle nested objects.